src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,498 @@
+import {
+  Component,
+  h,
+  Prop,
+  Element,
+  Event,
+  EventEmitter,
+  VNode,
+  Watch,
+  Method,
+  Host,
+  State
+} from "@stencil/core";
+import { connectForm, disconnectForm, FormComponent, HiddenFormInputSlot } from "../../utils/form";
+import { connectLabel, disconnectLabel, getLabelText, LabelableComponent } from "../../utils/label";
+import { slotChangeGetAssignedElements, toAriaBoolean } from "../../utils/dom";
+import { CSS, SLOTS, RESIZE_TIMEOUT } from "./resources";
+import {
+  connectLocalized,
+  disconnectLocalized,
+  LocalizedComponent,
+  NumberingSystem,
+  numberStringFormatter
+} from "../../utils/locale";
+import { createObserver } from "../../utils/observers";
+import {
+  componentLoaded,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent
+} from "../../utils/loadable";
+import {
+  connectMessages,
+  disconnectMessages,
+  setUpMessages,
+  T9nComponent,
+  updateMessages
+} from "../../utils/t9n";
+import { TextareaMessages } from "./assets/textarea/t9n";
+import { throttle } from "lodash-es";
+import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
+
+/**
+ * @slot - A slot for adding text.
+ * @slot footer-start - A slot for adding a leading footer.
+ * @slot footer-end - A slot for adding a trailing footer.
+ */
+
+@Component({
+  tag: "calcite-textarea",
+  styleUrl: "textarea.scss",
+  shadow: true,
+  assetsDirs: ["assets"]
+})
+export class Textarea
+  implements
+    FormComponent,
+    LabelableComponent,
+    LocalizedComponent,
+    LoadableComponent,
+    T9nComponent,
+    InteractiveComponent
+{
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTextareaElement;
```

Comment:
I always wonder why this variable isn't just named `host`. It always references the host so it makes sense to name it as host and it also reduces confusion later in the component because we know we are referring to the host and not some other random element.